workspaces/marketplace/plugins/marketplace-common/src/types.ts

```diff
@@ -93,3 +95,36 @@ export interface MarketplaceApi {
   getPluginListByName(name: string): Promise<MarketplacePluginList>;
   getPluginsByPluginListName(name: string): Promise<MarketplacePlugin[]>;
 }
+
+/**
+ * @public
+ */
+export interface MarketplaceAggregationApi {
+  fetchAggregatedData(
+    aggregationsRequest: AggregationsRequest,
+    baseQuery?: Knex.QueryBuilder | null,
```

Comment:
I didn't tested it yet, but the code and tests looks really nice. :+1:

This line (and the common dependency to knex) is the only aspect that I like to understand. I guess you would like to reuse this for tables other than final_entities. But I think the structure of final_entities is really "special", so it would surprise me if the rest would work with "another table". :thinking: 

Let us have a quick call later today.

Reply:
You are right!. It wouldn't work with another table. Initially I designed this function as recursive one so I was trying to build the query based on this optional param but later I changed it to iterative approach for simplicity and forgot to remove this code. Will remove this code and dependancy in both the places.